### PR TITLE
Migrate shopping-cart-service-cassandra-java

### PR DESCRIPTION
**Pull request by laglangyue**
## This PR inclueds:
 migrate `shopping-cart-service-cassandra-java` to pekko

## Test
Failed, there are some problems of configratution.
This module can be compiled with `mvn clean package -DskipTests=true`,
Due to lack of pekko-persistences of me and changes of configuration,I can't finish the test now.

### CODE REVIEW

**Comment by laglangyue** [comment]
I need more time to style the pekko-persistence and then finish the test.

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml [hunk 1]
```diff
@@ -15,99 +15,106 @@
...
         <!-- Version of the Docker image is derived from git commit.
              It can be defined with -Dversion.number=0.1-SNAPSHOT if git is not used -->
         <version.number>${git.commit.time}-${git.commit.id.abbrev}</version.number>
-        <!-- tag::akka-persistence-cassandra[] -->
+        <!-- tag::pekko-persistence-cassandra[] -->
     </properties>
-    <!-- end::akka-persistence-cassandra[] -->
+    <!-- end::pekko-persistence-cassandra[] -->
 
-    <dependencyManagement>
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml:33]: could you use the pekko-bom like I used in microservices tutorials for consistency? If you use that, then you don't need to set the pekko version on the jars that were `<groupId>com.typesafe.akka</groupId>`

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml [hunk 2]
```diff
@@ -75,8 +76,15 @@
         <dependency>
             <groupId>org.apache.pekko</groupId>
             <artifactId>pekko-discovery_${scala.binary.version}</artifactId>
-            <version>${pekko.version}</version>
+            <version>${pekko.discovery.version}</version>
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml:79]: this is wrong - pekko-discovery is part of the core pekko and should use v1.0.2

pekko-discovery-kubernetes-api is part of pekko-management and should use pekko-management.version

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml [hunk 3]
```diff
@@ -75,8 +76,15 @@
...
         </dependency>
+        <!-- https://mvnrepository.com/artifact/org.apache.pekko/pekko-discovery-kubernetes-api -->
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml:81]: remove this comment

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml (hunk 1 above)
Comment (laglangyue) [reply docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml:33]: I didn't find it similar at first.  Get it now, thanks.
          <dependency>
                <groupId>org.apache.pekko</groupId>
                <artifactId>pekko-bom_${scala.binary.version}</artifactId>
                <version>${pekko.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/docker-compose.yml [hunk 4]
```diff
@@ -10,7 +10,7 @@ services:
       timeout: 5s
       retries: 60
   # This exists to force the condition of having the Cassandra service is up before starting the tests.
-  # The healthcheck above is not enough because it does not provide a condition to wait for the service
+  # The health-check above is not enough because it does not provide a condition to wait for the service
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/docker-compose.yml:13]: Can you revert this? This is worse with a hyphen.

File: docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml [hunk 5]
```diff
@@ -187,9 +206,9 @@
             </plugin>
 
             <plugin>
-                <groupId>com.lightbend.akka.grpc</groupId>
-                <artifactId>akka-grpc-maven-plugin</artifactId>
-                <version>${akka-grpc-maven-plugin.version}</version>
+                <groupId>org.apache.pekko</groupId>
+                <artifactId>pekko-grpc-maven-plugin</artifactId>
+                <version>${pekko-grpc-maven-plugin.version}</version>
```
Comment (pjfanning) [inline docs-source/docs/modules/how-to/examples/shopping-cart-service-cassandra-java/pom.xml:211]: use pekko-grpc.version

**Comment by pjfanning** [comment]
I've merged in latest main code and fixed some issues.

There are 2 tests failing. I haven't had time to debug why.

**Comment by laglangyue** [comment]
> I've merged in latest main code and fixed some issues.
> 
> There are 2 tests failing. I haven't had time to debug why.
> 
> @laglangyue if you want to, you can try fixing this. I will try again in next few days. Please don't create any more code change PRs for this repo. It is easier to upgrade them altogether instead of 1 at a time. I did all the microservices-tutorials changes yesterday in 1 PR.

get it, I would like to test

**Pull request merged**